data/net/minecraft/BlockUtil.mapping

```diff
@@ -16,15 +16,28 @@ CLASS net/minecraft/BlockUtil
 	METHOD getMaxRectangleLocation ([I)Lcom/mojang/datafixers/util/Pair;
 		COMMENT Finds the largest rectangle within the array of heights
 		ARG 0 heights
+	METHOD getTopConnectedBlock (Lnet/minecraft/world/level/BlockGetter;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/Block;Lnet/minecraft/core/Direction;Lnet/minecraft/world/level/block/Block;)Ljava/util/Optional;
+		ARG 0 getter
+		ARG 1 pos
+		ARG 2 bottomBlock
+		ARG 3 direction
+		ARG 4 topBlock
```

Comment:
In most cases the direction is UP, but the WeepingVinesPlantBlock will call this method with DOWN. Because of that I'm not sure, why Mojang named the Method getTopConnectedBlock, but I would suggest to add a comment that this method is used for DOWN aswell(or that it can be used for other directions).
I think bottomBlock should be renamed to baseBlock and topBlock to endBlock or something similar